### ISSUE TITLE
TODO

### ISSUE THREAD

**Issue by pingrunhuang** [pingrunhuang/pingrunhuang.github.io#17]
- [ ] learn mockito test tutorial

**Comment by pingrunhuang** [comment]
# music-generation-with-DL

## Papers

- **Melody Generation for Pop Music via Word Representation of Musical Properties** (2017.10) [[arXiv](https://arxiv.org/abs/1710.11549)] [[Code](https://github.com/mil-tokyo/NeuralMelody)]

- **Generating Nontrivial Melodies for Music as a Service** (2017.10) [[arXiv](https://arxiv.org/abs/1710.02280)] [[Page](https://composing.ai)]

- **MuseGAN: Symbolic-domain Music Generation and Accompaniment with Multi-track Sequential Generative Adversarial Networks** (2017.9) [[arXiv](https://arxiv.org/abs/1709.06298)] [[Page](https://salu133445.github.io/musegan/)]

- **Similarity Embedding Network for Unsupervised Sequential Pattern Learning by Playing Music Puzzle Games** （2017.9）[[arXiv](https://arxiv.org/abs/1709.04384)] [[Page](https://remyhuang.github.io/DJnet)]

- **A Tutorial on Deep Learning for Music Information Retrieval** (2017.9) [[arXiv](https://arxiv.org/abs/1709.04396)]

- **Deep Learning Techniques for Music Generation - A Survey** (2017.9) [[arXiv](https://arxiv.org/abs/1709.01620)]

- **Neural Translation of Musical Style** (2017.8) [[arXiv](https://arxiv.org/abs/1708.03535)] [[Page](http://imanmalik.com/cs/2017/06/05/neural-style.html)]

- **GLSR-VAE: Geodesic Latent Space Regularization for Variational AutoEncoder Architectures** (2017.7) [[arXiv](https://arxiv.org/abs/1707.04588)]

- **Learning and Evaluating Musical Features with Deep Autoencoders** (2017.6) [[arXiv](https://arxiv.org/abs/1706.04486)]

- **Objective-Reinforced Generative Adversarial Networks (ORGAN) for Sequence Generation Models** (2017.5) [[arXiv](https://arxiv.org/abs/1705.10843)] [[Code](https://github.com/gablg1/ORGAN)]

- **MidiNet: A Convolutional Generative Adversarial Network for Symbolic-domain Music Generation using 1D and 2D Conditions** - **ISMIR 2017** (2017.3) [[arXiv](https://arxiv.org/abs/1703.10847)] [[Page](https://richardyang40148.github.io/TheBlog/midinet_arxiv_demo.html)] 

- **Automatic Conversion of Pop Music into Chiptunes for 8-bit Pixel Art** - **ICASSP 2017** (2017.2) [[Paper](http://mac.citi.sinica.edu.tw/~yang/pub/su17icassp_8bit.pdf)] [[Code](https://github.com/LemonATsu/pop-to-8bit)] [[Page](https://lemonatsu.github.io)]


- **C-RNN-GAN: Continuous Recurrent Neural Networks with Adversarial Training** (2016.11) [[arXiv](https://arxiv.org/abs/1611.09904)] [[Code](https://github.com/olofmogren/c-rnn-gan)]

- **Tuning Recurrent Neural Networks with Reinforcement Learning - ICLR 2017** (2016.11) [[arXiv](https://arxiv.org/abs/1611.02796)] [[Web](https://magenta.tensorflow.org/2016/11/09/tuning-recurrent-networks-with-reinforcement-learning)] [[Code](https://github.com/tensorflow/magenta/tree/master/magenta/models/rl_tuner)]

- **SeqGAN: Sequence Generative Adversarial Nets with Policy Gradient** - **AAAI 2017** (2016.9) [[Paper](http://www.aaai.org/ocs/index.php/AAAI/AAAI17/paper/download/14344/14489)] [[Code](https://github.com/LantaoYu/SeqGAN)]

- **Song From PI: A Musically Plausible Network for Pop Music Generation** - **ICLR 2017** [[arXiv](https://arxiv.org/abs/1611.03477)]

- **Text-based LSTM networks for Automatic Music Composition** (2016.4) [[arXiv](https://arxiv.org/abs/1604.05358#)] [[Web](https://keunwoochoi.wordpress.com/2016/02/23/lstmetallica/)] [[Code](https://github.com/keunwoochoi/LSTMetallica)]

- **Music Transcription Modelling and Composition Using Deep Learning** (2016.4) [[arXiv](https://arxiv.org/abs/1604.08723)] [[Code](https://github.com/IraKorshunova/folk-rnn)]

- **Composing A Melody with Long-short Term Memory (LSTM) Recurrent Neural Networks** (2016.2) [[Web](http://konstilackner.github.io/LSTM-RNN-Melody-Composer-Website/)] [[Code](https://github.com/konstilackner/LSTM-RNN-Melody-Composer)] [[Paper](http://konstilackner.github.io/LSTM-RNN-Melody-Composer-Website/Thesis_final01.pdf)]

- **Neural Adaptive Sequential Monte Carlo - NIPS 2015** (2015) [[Paper](http://papers.nips.cc/paper/5961-neural-adaptive-sequential-monte-carlo.pdf)]

- **A Recurrent Latent Variable Model for Sequential Data - NIPS 2015** (2015) [[Paper](http://papers.nips.cc/paper/5653-a-recurrent-latent-variable-model-for-sequential-data.pdf)] [[Code](https://github.com/jych/nips2015_vrnn)]

- **AI Methods in Algorithmic Composition: A Comprehensive Survey** (2013) [[Paper](http://www.jair.org/media/3908/live-3908-7454-jair.pdf)]

- **Modeling Temporal Dependencies in High-dimensional Sequences: Application to Polyphonic Music Generation and Transcription** (2012) [[arXiv](https://arxiv.org/abs/1206.6392)] 

- **Towards Adaptive Music Generation By Reinforcement Learning of Musical Tension** (2010) [[Paper](https://ccrma.stanford.edu/~slegroux/affect/pubs/SMC2010.pdf)]

- **A First Look at Music Composition using LSTM Recurrent Neural Networks** (2002) [[Web](http://www.iro.umontreal.ca/~eckdoug/blues/index.html)] [[Paper](http://www.iro.umontreal.ca/~eckdoug/blues/IDSIA-07-02.pdf)]

## Blogs

- **Neural Nets for Generating Music** [[Web](https://medium.com/@kcimc/neural-nets-for-generating-music-f46dffac21c0)]

- **Generative Music with JavaScript & Web Audio** [[Web](https://teropa.info/generative-music-slides/)]

- **The Current State Of AI: Artificial Intelligence In Music, Movies & More** (2017.7) [[Web](http://www.hypebot.com/hypebot/2017/07/ai-today-the-current-state-of-artificial-intelligence.html)]
- **Composing Music With Recurrent Neural Networks** (2015.8) [[Web](http://www.hexahedria.com/2015/08/03/composing-music-with-recurrent-neural-networks/)] [[Code](https://github.com/hexahedria/biaxial-rnn-music-composition)]


## Codes

-  **Google Magenta** [[Web](https://magenta.tensorflow.org/welcome-to-magenta)] [[Code](https://github.com/tensorflow/magenta)] 

- **Deep Jazz**  [[Web](https://deepjazz.io/)] [[Code](https://deepjazz.io/)]

- **BachBot** [[Web](http://bachbot.com/)] [[Code](https://github.com/feynmanliang/bachbot/)]

- **WaveNet** [[Code](https://github.com/ibab/tensorflow-wavenet)] (not fully)

- **GRUV** [[Code](https://github.com/MattVitelli/GRUV)]

- **Kulitta** [[Code](https://github.com/donya/Kulitta)]

## Conferences&Workshops

- **ACM MM** - ACM MultiMedia [[Web](http://www.acmmm.org/2017)]
- **ISMIR** - The International Society of Music Information Retrieval [[Web](http://www.ismir.net/)]
- **ICASSP** - Conference on Acoustics, Speech and Signal Processing [[Web](http://www.ieee-icassp2017.org/)]
- **DLM** - Deep Learning for Music Workshop [[Web](http://dorienherremans.com/dlm2017/)]
- **CSMC** - Conference on Computer Simulation of Musical  Creativity [[Web](https://csmc2016.wordpress.com/)]
- **CCRMA** - Center for Computer Research in Music and Acoustics (Stanford University) [[Web](https://ccrma.stanford.edu/)]
- **ICMC** - Internatonal Computer Music Conference [[Web](http://www.icmc2017.com/)] [[Lists](http://www.icmc2017.com/cn/page1.html)]

## Applications

- **Google A.I. Duet** [[Link](https://aiexperiments.withgoogle.com/ai-duet)]
- **The Infinite Drum Machine** [[Link](https://aiexperiments.withgoogle.com/drum-machine)]
- **Amper Music** [[Link](https://www.ampermusic.com/app#/)]
- **Intelligent Music System** [[Link](http://120.52.72.53/www.intelligentmusicsystems.com/c3pr90ntc0td/vid/tempo_shifting.mp4)]
- **Unwind** [[Link](http://unwind.ai)]
- **Tidalcycles** [[Link](https://tidalcycles.org)] [[Video](https://www.youtube.com/watch?v=xoa3OT8ncX0)]

**Comment by pingrunhuang** [comment]
**2018-07-10**
- [ ] 完成closure博文
- [ ] 关于谷歌应聘的问题回答
- [ ] 做最近项目的难点分析和应对策略，数据很重要
- [ ] 学习scala多线程
